src/controller/CHIPDeviceController.cpp

```diff
@@ -1487,6 +1487,8 @@ void DeviceCommissioner::CommissioningStageComplete(CHIP_ERROR err, Commissionin
 
     if (mPairingDelegate != nullptr)
     {
+        // TODO github issue mentions that the intent it to plumb result to PairingDelegate, but over
```

Comment:
Both - the CommissioningDelegate (AutoCommissioner is the example CommissiningDelegate) handles the sequencing of the stages, so it needs to know that this stage was an error. It should also be reported out to the PairingDelegate, which gets status reports for each stage as well as Success/Failure reports. The CommissioningDelegate can use the information in the CommissioningReport to set the CompletionStatus in the CommissioningParameters. That's what gets sent to the PairingDelegate when Cleanup is called and the final Success or Completion notification is sent.

Reply:
Okay so that wouldn't be done with `mPairingDelegate` here, but instead it would already be done with what I have inside `AutoCommissioner::CommissioningStepFinished` when `mParams.SetCompletionStatus(completionStatus);` is called. Is that correct?

Reply:
Yup - that's correct.